File: client/src/Components/Tags/TagData.ts
export interface TagData {
  id: string
  name: string
  slug: string
  num_posts: number
  created_at: string
  updated_at: string
}

type EditableFields =
  | 'name'
  | 'slug'

export type EditableTagData = Pick<TagData, EditableFields>

File: packages/og-injector-rust/src/server/models/tag_data.rs
use iso8601::DateTime;
use serde::Deserialize;
use uuid::Uuid;

#[derive(Deserialize)]
#[allow(dead_code)]
pub struct TagData {
    id: Uuid,
    name: String,
    slug: String,
    num_posts: u64,
    #[serde(with = "crate::serde::date_time")]
    created_at: DateTime,
    #[serde(with = "crate::serde::date_time")]
    updated_at: DateTime,
    featured: bool,
}
